Sources/SwiftUIBackports/Shared/StateObject/StateObject.swift

```diff
@@ -0,0 +1,111 @@
+import Combine
+import SwiftUI
+
+extension Backport where Content == Any {
+
+    /// A property wrapper type that instantiates an observable object.
+    @propertyWrapper public struct StateObject<ObjectType: ObservableObject>: DynamicProperty {
+        private final class Wrapper: ObservableObject {
+            private var subject = PassthroughSubject<Void, Never>()
+
+            var value: ObjectType? {
+                didSet {
+                    cancellable = nil
+                    cancellable = value?.objectWillChange
+                        .sink { [subject] _ in subject.send() }
+                }
+            }
+
+            private var cancellable: AnyCancellable?
+
+            var objectWillChange: AnyPublisher<Void, Never> {
+                subject.eraseToAnyPublisher()
+            }
+        }
+
+        @State private var state = Wrapper()
+
+        @ObservedObject private var observedObject = Wrapper()
+
+        private var thunk: () -> ObjectType
```

Comment:
What does `thunk` mean? Sorry, never seen that term before.

Reply:
I'd prefer `closure` perhaps unless you have a better suggestion? But `thunk` is pretty obscure.

Reply:
I've seen it used in Swift compiler and with code generated for Xcode previews. As I understand it, it means a closure that wraps some code for later execution.
The naming comes from the argument on the official `@StateObject` implementation, so my preference would be to keep it. https://developer.apple.com/documentation/swiftui/stateobject/init(wrappedvalue:)